draft-ietf-quic-transport.md

```diff
@@ -785,13 +785,12 @@ flow control limits.
 If a sender runs out of flow control credit, it will be unable to send new data
 and is considered blocked.  A sender SHOULD send a STREAM_DATA_BLOCKED or
 DATA_BLOCKED frame to indicate it has data to write but is blocked by flow
-control limits.  These frames are expected to be sent infrequently in common
-cases, but they are considered useful for debugging and monitoring purposes.
-
-A sender SHOULD NOT send multiple STREAM_DATA_BLOCKED or DATA_BLOCKED frames
-for the same data limit, unless the original frame is determined to be lost.
-Another STREAM_DATA_BLOCKED or DATA_BLOCKED frame can be sent after the data
-limit is increased.
+control limits.  If a sender is blocked for long enough it is possible for the
+connection to idle timeout, even though the data is actively queued to be sent.
+To prevent this idle timeout from occurring, the sender SHOULD continue to
+periodically send the STREAM_DATA_BLOCKED or DATA_BLOCKED frame.  One method is
+to immediately resend the frame on acknowledgment of the previous one if the
```

Comment:
I tend to think this might be overly aggressive?  If 10 streams are blocked, there's no reason to send 10 frames an RTT, is there?

Reply:
I'm not trying to make any requirements on how many frames and how often you must send them. The only SHOULD here is to periodically (less than idle timeout) send some blocked frames. I'd leave it up to implementations to decide exactly what to do here. How should the text change (if at all) to make that more clear?

Reply:
How about "One method is to resend a DATA_BLOCKED or STREAM_DATA_BLOCKED frame when there are no ack-eliciting packets in flight."